x-pack/packages/kbn-elastic-assistant/impl/assistant/use_conversation/index.tsx

```diff
@@ -51,299 +48,264 @@ interface CreateConversationProps {
-        } else {
-          return prev;
-        }
+      await updateConversationApi({
+        http,
+        conversationId,
+        apiConfig: {
+          defaultSystemPromptId,
+        },
+        messages: [],
+        replacements: undefined,
       });
     },
-    [allSystemPrompts, setConversations]
+    [allSystemPrompts, http]
   );
 
   /**
    * Create a new conversation with the given conversationId, and optionally add messages
    */
-  const createConversation = useCallback(
+  const getDefaultConversation = useCallback(
     ({ conversationId, messages }: CreateConversationProps): Conversation => {
       const defaultSystemPromptId = getDefaultSystemPrompt({
         allSystemPrompts,
         conversation: undefined,
       })?.id;
 
-      const newConversation: Conversation = {
-        ...DEFAULT_CONVERSATION_STATE,
-        apiConfig: {
-          ...DEFAULT_CONVERSATION_STATE.apiConfig,
-          defaultSystemPromptId,
-        },
-        id: conversationId,
-        messages: messages != null ? messages : [],
-      };
-      setConversations((prev: Record<string, Conversation>) => {
-        const prevConversation: Conversation | undefined = prev[conversationId];
-        if (prevConversation != null) {
-          throw new Error('Conversation already exists!');
-        } else {
-          return {
-            ...prev,
-            [conversationId]: {
-              ...newConversation,
-            },
-          };
-        }
-      });
+      const newConversation: Conversation =
+        conversationId === i18n.WELCOME_CONVERSATION_TITLE
+          ? WELCOME_CONVERSATION
+          : {
+              ...DEFAULT_CONVERSATION_STATE,
+              apiConfig: {
+                ...DEFAULT_CONVERSATION_STATE.apiConfig,
+                defaultSystemPromptId,
+              },
+              id: conversationId,
+              messages: messages != null ? messages : [],
+            };
       return newConversation;
     },
-    [allSystemPrompts, setConversations]
+    [allSystemPrompts]
   );
 
   /**
-   * Delete the conversation with the given conversationId
+   * Create a new conversation with the given conversation
    */
-  const deleteConversation = useCallback(
-    (conversationId: string): Conversation | undefined => {
-      let deletedConversation: Conversation | undefined;
-      setConversations((prev: Record<string, Conversation>) => {
-        const { [conversationId]: prevConversation, ...updatedConversations } = prev;
-        deletedConversation = prevConversation;
-        if (prevConversation != null) {
-          return updatedConversations;
-        }
-        return prev;
-      });
-      return deletedConversation;
+  const createConversation = useCallback(
+    async (conversation: Conversation): Promise<Conversation | undefined> => {
+      const response = await createConversationApi({ http, conversation });
+      if (!isHttpFetchError(response)) {
+        return response;
```

Comment:
Consider displaying an error toast indicating the conversation will not be persisted